src/components/features/ProfileMenu.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react'
+import Button from '@mui/material/Button'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import { Box } from '@mui/material'
+import Dialog from '@mui/material/Dialog'
+import DialogActions from '@mui/material/DialogActions'
+import DialogContent from '@mui/material/DialogContent'
+import DialogContentText from '@mui/material/DialogContentText'
+import DialogTitle from '@mui/material/DialogTitle'
+import { useTranslation } from 'react-i18next'
+
+export default function ProfileMenu() {
+  const { t: translate } = useTranslation()
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const menuOpen = Boolean(anchorEl)
+  const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const handleMenuClose = () => {
+    setAnchorEl(null)
+  }
+
+  const [dialogOpen, setDialogOpen] = React.useState(false)
+
+  const handleDialogOpen = () => {
+    setDialogOpen(true)
+  }
+
+  const handleDialogClose = () => {
+    setDialogOpen(false)
+    handleMenuClose()
+  }
+
+  return (
+    <>
+      <Box
+        component="div"
+        sx={{
+          fontSize: '1.2em'
+        }}>
+        {translate('profileMenu.name')}
+      </Box>
+
+      <Button
+        id="basic-button"
+        aria-controls={menuOpen ? 'basic-menu' : undefined}
+        aria-haspopup="true"
+        aria-expanded={menuOpen ? 'true' : undefined}
+        onClick={handleMenuClick}>
+        <KeyboardArrowDownIcon sx={{ color: 'black' }} />
+      </Button>
+      <Menu
+        id="basic-menu"
+        anchorEl={anchorEl}
+        open={menuOpen}
+        onClose={handleMenuClose}
+        MenuListProps={{
+          'aria-labelledby': 'basic-button'
+        }}>
+        <MenuItem onClick={handleDialogOpen}>
+          <Box
+            component="div"
+            sx={{
+              textDecoration: 'none',
+              color: 'black'
+            }}>
+            {translate('profileMenu.labels.exit')}
+          </Box>
+        </MenuItem>
+      </Menu>
+
+      <Dialog
```

Comment:
lets make this into its own component, for example `WarningActionDialog` and the props can look something like this

```ts
// we can maybe have the confirmText props as optional so maybe the component can be used just for info warning
interface WarningActionDialogProps { 
      open: boolean
      title: string
      content?: string
      discardText: string
      confirmText?: string
      onClose: () => void
      onDiscardClick: () => void
      onConfirmClick?: () => void
}
```

Reply:
Fixed